index.js

```diff
@@ -189,4 +189,31 @@ function handleFormSubmit(event) {
   }
 }
 
-document.getElementById('contact-form').addEventListener('submit', handleFormSubmit);
\ No newline at end of file
+document.getElementById('contact-form').addEventListener('submit', handleFormSubmit);
+
+function saveData() {
+  const name = document.getElementById('name').value;
+  const email = document.getElementById('email').value;
+  const message = document.getElementById('msg').value;
+
+  const formData = {
+    name,
+    email,
+    message,
+  };
+  localStorage.setItem('formData', JSON.stringify(formData));
+
+  const xhr = new XMLHttpRequest();
+  xhr.open('POST', 'https://formspree.io/f/mnqygqol');
+  xhr.setRequestHeader('Content-Type', 'application/json');
+  xhr.send(JSON.stringify(formData));
+}
+
+const form = document.getElementById('contact-form');
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+  if (form.checkValidity() && validateEmail()) {
+    saveData();
+    form.reset();
+  }
+});
```

Comment:
- [ ] Great job so far @mumbereh However, I noticed that data is not being preserved in your form as expected :+1: 
- [ ] Per the project requirements, after filling out the form, and not submitting it, the entered values should be saved in the browser but currently, they are only saved once the form is submitted and even with that the contact form information isn't persistent to page reloads as a result the form is still empty even though the `formdata` object was saved in the local storage so kindly fix this. :+1:  
- [ ] Please also ensure that when a user changes the content of any input field, the changes are also saved into the local storage :100: 

[Screencast from 06-21-2023 11:41:05 AM.webm](https://github.com/mumbereh/my-portfolio/assets/102699958/9a9b9204-21e3-428e-96d0-798ebb798c85)